rmw_connext_cpp/src/rmw_subscription.cpp

```diff
@@ -365,6 +365,96 @@ rmw_subscription_count_matched_publishers(
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+rmw_subscription_get_actual_qos(
+  const rmw_subscription_t * subscription,
+  rmw_qos_profile_t * qos)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(subscription, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_ARGUMENT_FOR_NULL(qos, RMW_RET_INVALID_ARGUMENT);
+
+  auto info = static_cast<ConnextStaticSubscriberInfo *>(subscription->data);
+  if (!info) {
+    RMW_SET_ERROR_MSG("subscription internal data is invalid");
+    return RMW_RET_ERROR;
+  }
+  DDS::DataReader * data_reader = info->topic_reader_;
+  if (!data_reader) {
+    RMW_SET_ERROR_MSG("subscription internal data reader is invalid");
+    return RMW_RET_ERROR;
+  }
+  DDS::DataReaderQos dds_qos;
+  DDS::ReturnCode_t status = data_reader->get_qos(dds_qos);
+  if (DDS::RETCODE_OK != status) {
+    RMW_SET_ERROR_MSG("subscription can't get data reader qos policies");
+    return RMW_RET_ERROR;
+  }
+
+  if (!data_reader) {
```

Comment:
Why is this check needed? I think the one in L382 is enough. I know that the same was done in:
https://github.com/ros2/rmw_connext/blob/8e9e08825c91b0a7abd9ef04028bbb515a3105f4/rmw_connext_cpp/src/rmw_publisher.cpp#L377-L393
But I don't think it has sense. If you agree, delete the duplicated check in the publisher function too

Reply:
Done.